run-vmtest/print_test_summary.py

```
@@ -75,31 +75,41 @@ def test_error_console_log(test_error: str, test_message: str) -> str:
     else:
         return error_msg
 
+def report_error(msg: str):
+    print("print_test_summary.py: {}".format(msg), file=sys.stderr)
 
 if __name__ == "__main__":
     args = parse_args()
     step_open_mode = "a" if args.append else "w"
     json_summary = None
 
-    with open(args.json_summary, "r") as f:
-        json_summary = json.load(f)
-
-    with open(args.step_summary, step_open_mode) as f:
-        log_gh_summary(f, "# Tests summary")
-        log_gh_summary(f, markdown_summary(json_summary))
-
-        log_console(notice(console_summary(json_summary)))
-
-        for test in json_summary["results"]:
-            test_name = test["name"]
-            test_number = test["number"]
-            if test["failed"]:
-                test_log = f"#{test_number} {test_name}"
-                log_gh_summary(f, test_log)
-                log_console(test_error_console_log(test_log, test["message"]))
-
-            for subtest in test["subtests"]:
-                if subtest["failed"]:
-                    subtest_log = f"#{test_number}/{subtest['number']} {test_name}/{subtest['name']}"
-                    log_gh_summary(f, subtest_log)
-                    log_console(test_error_console_log(subtest_log, subtest["message"]))
+    try:
+        with open(args.json_summary, "r") as f:
+            json_summary = json.load(f)
+    except:
+        report_error("Could not load JSON summary file {}".format(args.json_summary))
```

Is there a way to get actual parsing error from the exception?

---

It's usually `JSONDecodeError` that we could catch, but then any other failure would result in the traceback splat.

I think it's not important to distinguish between errors within this script. It only fails if something went wrong with the tests.

---

Is there still a way to infer why operation failed? As far as I understand, this completely hides exception.

---

> Is there still a way to infer why operation failed? 

If you mean the script itself, then yeah, we could handle different exceptions differently. But I don't see much benefit to that. The reason for this change is to reduce noise. "File not found" or "summary file is empty" error  doesn't help in the context of CI.

If you mean "can we print a useful tests summary even if json was not generated?", that's a different problem. I think that would require capturing stdout/stderr of the vmtest program.

---

No, I mean, suppose something is wrong with json printing or the file is missing, would there be a log anywhere saying that json is malformed or absent, or "could not load" is all you'd get?

---

> No, I mean, suppose something is wrong with json printing or the file is missing, would there be a log anywhere saying that json is malformed or absent, or "could not load" is all you'd get?

The way the change is written you'll only see the "could not load" message, yes. I want to get rid of useless traceback we often see in the job log.

Are you worried that we may miss a malformed json produced by a test runner? Has that ever happened?

---

> Are you worried that we may miss a malformed json produced by a test runner? Has that ever happened?

No, just would be harder to figure out what happened. Point being, just add `str(e)` to the error string?

---

I changed to handling "expected" errors: no file, empty file, json decode error. Anything else should produce a stacktrace. Thanks for taking a look.